### PR TITLE
Change the redux actions/reducers to retrieve survey, surveys metadata, and save survey.

### PR DESCRIPTION
**Pull request by ayushbhagat**
Solves uwblueprint/somo#63.

### CODE REVIEW

**Comment by hkjayakumar** [comment]
@ayushbhagat Can you make it pass CI ?

**Comment by ayushbhagat** [comment]
That's because I haven't changed the tests. It will once I change the tests. Can you take a look at the source code that I have changed and leave comments if there are any.

**Comment by hkjayakumar** [comment]
Ohh alright, will do

**Review by hkjayakumar — COMMENTED** [review]
LGTM !

Will approve after tests are written and CI passes.

File: test/actions/surveys.js
Comment (hkjayakumar) [inline test/actions/surveys.js]: Nit: 'should dispatch `an` action'

File: test/actions/surveys.js
Comment (hkjayakumar) [inline test/actions/surveys.js:35]: Why is this in quotes ?

File: test/actions/surveys.js
Comment (hkjayakumar) [inline test/actions/surveys.js:32]: Nit: 'should dispatch `an` action' (or change line 15 to make it consistent)

File: test/actions/surveys.js
Comment (hkjayakumar) [inline test/actions/surveys.js:49]: Consistency

File: src/actions/surveys.js
Comment (hkjayakumar) [inline src/actions/surveys.js:23]: Can you add a comment in the commit message as to why its not longer Immutable when merging the PR ?

**Review by hkjayakumar — CHANGES_REQUESTED** [review]
1) Fix conflicts.
2) Consistency of messages.
3) Mention why we removed Immutable just before merging.
4) Why is the survey object now split into a survey id and survey body ?

File: src/actions/surveys.js
Comment (ayushbhagat) [reply src/actions/surveys.js:23]: It's because I changed the actions and reducers so that the actions are a JavaScript object and the reducers are the ones that convert it to a Immutable object. This makes more sense because only the data that is stored in the state needs to be immutable; the actions don't. How about instead of writing that commit message, I add the explanation for when to use immutable in the README styles section?

File: test/actions/surveys.js
Comment (ayushbhagat) [reply test/actions/surveys.js:32]: Done. I removed the `an` from line 15.

File: test/actions/surveys.js
Comment (ayushbhagat) [reply test/actions/surveys.js:35]: `id` can be the string `new` when the user is on the page to create a new survey. So, for consistency, in the redux state, I store the `id` for all surveys as a string.

File: test/actions/surveys.js
Comment (ayushbhagat) [reply test/actions/surveys.js:49]: Done.

**Comment by ayushbhagat** [comment]
@hkjayakumar PTAL. I made the requested changes.

**Review by hkjayakumar — APPROVED** [review]

File: test/actions/surveys.js
Comment (chrisjluc) [reply test/actions/surveys.js:35]: Maybe having  `-1` is better than `'new'`, that way it can still be an int

Comment (ayushbhagat) [reply test/actions/surveys.js:35]: So the reason why I made it `'new'` was because I wanted to ensure that the route for the page where you can edit a survey with an id `id` was `/survey/id`. And since the route for the page where you can create a survey is called `/survey/new`, I made the id `'new'`. But now that I think about it, I don't think having the same `id` as one in the route provides any benefit for a survey that hasn't been persisted. So I'll change it.

**Review by chrisjluc — APPROVED** [review]

**Comment by chrisjluc** [comment]
@hkjayakumar can you review this again?

**Review by hkjayakumar — APPROVED** [review]

**Pull request merged**
